src/GHC/SourceGen/Syntax/Internal.hs

```diff
@@ -88,6 +91,9 @@ noExtOrPlaceHolder = withPlaceHolder
 withPlaceHolder :: (PlaceHolder -> a) -> a
 withPlaceHolder = ($ PlaceHolder)
 
+withPlaceHolders :: ([PlaceHolder] -> a) -> a
+withPlaceHolders = ($ [])
```

Comment:
I'm assuming putting `[]` here is fine because its only being used after type checking phase.

Reply:
Yeah, this approach seems reasonable to me.